### ISSUE TITLE
`@sveltejs/package` missing `import` in `package.json.exports['.']`?

### ISSUE THREAD

**Issue by janosh** [sveltejs/vite-plugin-svelte#710]
### Describe the bug

[I started running into](https://github.com/janosh/elementari/actions/runs/5791065675/job/15695238618?pr=42)

> Error: Failed to resolve entry for package "@sveltejs/package". The package may have incorrect main/module/exports specified in its package.json: No known conditions for "." specifier in "@sveltejs/package" package

<details><summary>Stacktrace</summary>

```
❯ packageEntryFailure node_modules/vite/dist/node/chunks/dep-df561101.js:28691:11
 ❯ resolvePackageEntry node_modules/vite/dist/node/chunks/dep-df561101.js:28686:9
 ❯ tryNodeResolve node_modules/vite/dist/node/chunks/dep-df561101.js:28419:20
 ❯ Context.resolveId node_modules/vite/dist/node/chunks/dep-df561101.js:28[18](https://github.com/janosh/elementari/actions/runs/5791065675/job/15695238618?pr=42#step:5:19)0:28
 ❯ Object.resolveId node_modules/vite/dist/node/chunks/dep-df561101.js:44207:64
 ❯ async file:/home/runner/work/elementari/elementari/node_modules/vite/dist/node/chunks/dep-df561101.js:65837:21
 ❯ async file:/home/runner/work/elementari/elementari/node_modules/vite/dist/node/chunks/dep-df561101.js:44852:[20](https://github.com/janosh/elementari/actions/runs/5791065675/job/15695238618?pr=42#step:5:21)
 ❯ addManuallyIncludedOptimizeDeps node_modules/vite/dist/node/chunks/dep-df561101.js:46034:31
 ❯ optimizeServerSsrDeps node_modules/vite/dist/node/chunks/dep-df561101.js:45638:5
 ❯ createDevSsrDepsOptimizer node_modules/vite/dist/node/chunks/dep-df561101.js:45556:[22](https://github.com/janosh/elementari/actions/runs/5791065675/job/15695238618?pr=42#step:5:23)
```
</details> 

I'm no packaging expert but looking at

https://github.com/sveltejs/kit/blob/a6fe5fcb1c7258281b4bf53b94543272e6e6c6d8/packages/package/package.json#L44-L50

isn't that missing an `"import": "./src/exports/index.js"` or similar (going by `kit` exports).

https://github.com/sveltejs/kit/blob/a6fe5fcb1c7258281b4bf53b94543272e6e6c6d8/packages/kit/package.json#L75-L100

### Reproduction

Tried but failed to repro on StackBlitz.

### System Info

```Shell
System:
    OS: macOS 13.4
    CPU: (12) arm64 Apple M2 Max
    Memory: 4.25 GB / 64.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.5.0 - /opt/homebrew/bin/node
    npm: 9.8.0 - /opt/homebrew/bin/npm
    pnpm: 8.6.2 - /opt/homebrew/bin/pnpm
  Browsers:
    Brave Browser: 115.1.56.20
    Chrome: 115.0.5790.170
    Safari: 16.5
  npmPackages:
    @sveltejs/adapter-static: 2.0.3 => 2.0.3 
    @sveltejs/kit: ^1.22.4 => 1.22.4 
    @sveltejs/package: ^2.2.0 => 2.2.0 
    svelte: 4.1.2 => 4.1.2 
    vite: ^4.4.9 => 4.4.9
```

### Severity

blocking all usage of SvelteKit

**Comment by dominikg** [comment]
moved the issue to vite-plugin-svelte.

It looks like`vitefu` adds `@sveltejs/package` and `svelte2tsx` to `ssr.noExternal`

```
 vite:vite-plugin-svelte extra config for dependencies generated by vitefu {
  optimizeDeps: {
    include: [
      '@sveltejs/package > chokidar',
      '@sveltejs/package > semver',
      '@sveltejs/package > svelte2tsx > dedent-js'
    ],
    exclude: []
  },
  ssr: {
    noExternal: [ '@sveltejs/package', 'svelte2tsx' ],
    external: [
      'chokidar',
      'kleur',
      'sade',
      'semver',
      'dedent-js',
      'pascal-case'
    ]
  }
}
```

and vite adds packages present in ssr.noExternal to optimizeDeps.include.  

We may have to update some config or change the filter functions we use with vitefu to prevent these from showing up. I'm also a bit confused about some of the values in ssr.external.

**Comment by bluwy** [comment]
If the error is coming from Vitest [0.34](https://github.com/vitest-dev/vitest/releases/tag/v0.34.0) specifically, it's likely because that version started to use `ssr.external` and `ssr.noExternal`, instead of `deps.external` and `deps.inline`, that's causing these errors. Vite's `ssr` configs are usually lazy, so setting these packages incorrectly is fine as long as you don't import it, but Vitest seems to made it eager (which I don't exactly think it's a bug there).

The PR fix sent is probably the simplest way to fix this though.

**Comment by dominikg** [comment]
released fix in 2.4.5  please let us know if you still have this issue aftert updating

**Comment by oganexon** [comment]
> released fix in 2.4.5 please let us know if you still have this issue aftert updating

I'm having the same issue :
```
Failed to resolve entry for package "sveltekit-flash-message". The package may have incorrect main/module/exports specified in its package.json: No known conditions for "." specifier in "sveltekit-flash-message" package
```

Packages :
```json
{
    "@sveltejs/kit": "1.25.1",
    "svelte": "4.2.1",
    "sveltekit-flash-message": "2.2.1",
    "vite": "4.4.10",
}
```
